### DISCUSSION TITLE
Magic numbers in `serializer.WriteProperty` and `deserializer.ReadProperty`

### DISCUSSION THREAD

**Discussion by PeiMu** [duckdb/duckdb#11775] [General]
### What happens?

Hi,

I'm not sure if this is the proper place to ask this. (If not, I can definitely move)

When I was implementing my code based on duckdb, I found it a bit confusing about the magic numbers in the `src/storage/serialization/serialize_logical_operator.cpp`.

For example, if I added a member `bool split_point = false;` in `class LogicalOperator`, I found when I needed to copy the `LogicalOperator`, I went to the `Serialize` and `Deserialize` process. Can I use a continuing number, e.g. 102?

```c++
void LogicalOperator::Serialize(Serializer &serializer) const {
	serializer.WriteProperty<LogicalOperatorType>(100, "type", type);
	serializer.WritePropertyWithDefault<vector<unique_ptr<LogicalOperator>>>(101, "children", children);
	serializer.WriteProperty<bool>(102, "split_point", split_point);
}

unique_ptr<LogicalOperator> LogicalOperator::Deserialize(Deserializer &deserializer) {
	auto type = deserializer.ReadProperty<LogicalOperatorType>(100, "type");
	auto children = deserializer.ReadPropertyWithDefault<vector<unique_ptr<LogicalOperator>>>(101, "children");
	auto split_point = deserializer.ReadProperty<bool>(102, "split_point");
```

Many thanks,
Pei

### To Reproduce

NULL

### OS:

linux

### DuckDB Version:

0.10.0-4a89d97

### DuckDB Client:

CLI

### Full Name:

Pei Mu

### Affiliation:

University of Edinburgh

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a source build

### Did you include all relevant data sets for reproducing the issue?

No - Other reason (please specify in the issue body)

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by Tishj** [comment]
See https://github.com/duckdb/duckdb/pull/8863
